### DISCUSSION TITLE
How to make any physical turn be considered as an actual navigation turn

### DISCUSSION THREAD

**Discussion by mecool1** [osmandapp/OsmAnd#18527] [Q&A] [answered]
I generated a bike route, which was full of turns on different types of pavements (road, gravel, narrow hiking trails, etc). Many of these turns clearly 90 degrees, but many of them did not trigger an arrow during navigation for various reasons. Typically, you are on a gravel road that takes a sharp right at a crossroad where the 3 other possibilities are smaller trails and it's not considered as a turn, but you still need to doublecheck, because with a mountain bike you could take all 4 paths.

In other words, I'm looking for a way to make any physical turn beyond a certain number of degrees be considered as an actual navigational turn (or maybe waypoint).

It would be especially useful in natural areas where minor trails criss-cross all over the place.

Ultimately, my goal is to leave my phone in my pocket while I cycle and receive voice prompts at every turn.

**Comment by mikehgentry** [comment, marked as answer]
If you import a GPX and *don't* use the 'attach to the roads' option you should get something like that I think

**Comment by DmitryAlexei** [comment]
Use attach to roads option. For more information see https://osmand.net/docs/user/navigation/setup/gpx-navigation#attach-to-roads

**Comment by mecool1** [comment]
OK, it works perfectly as long as the Attach to roads option is not turned on.

Thanks
